primary/src/block_synchronizer/mod.rs

```diff
@@ -605,13 +605,16 @@ impl BlockSynchronizer {
     // It returns back the primary names to which we have sent the requests.
     #[instrument(level = "debug", skip_all)]
     async fn broadcast_batch_request(&mut self, message: PrimaryMessage) -> Vec<PublicKey> {
-        // Naively now just broadcast the request to all the primaries
-
+        // Naively now just broadcast the request to all the primaries we have
+        // addresses for.
         let (primaries_names, primaries_addresses) = self
             .committee
             .others_primaries(&self.name)
             .into_iter()
-            .map(|(name, address)| (name, address.primary_to_primary))
+            .filter(|(_, address)| address.is_some())
+            .map(|(name, address)| (name, address.unwrap().primary_to_primary))
+            .filter(|(_, primary_to_primary)| primary_to_primary.is_some())
+            .map(|(name, primary_to_primary)| (name, primary_to_primary.unwrap()))
```

Comment:
This is more concisely:
```suggestion
            .filter_map(|(name, address)| address.map(|addr| addr.primary_to_primary.map(|p2p| (name, p2p))))
```
It's fine to decompose this into more matching for clarity, rather than using combinators, but I think it's both possible and worthwhile to avoid unwraps.